test/e2e/scheduling/limit_range.go

```diff
@@ -63,21 +74,48 @@ var _ = SIGDescribe("LimitRange", func() {
 			LabelSelector:   selector.String(),
 			ResourceVersion: limitRanges.ListMeta.ResourceVersion,
 		}
-		w, err := f.ClientSet.CoreV1().LimitRanges(f.Namespace.Name).Watch(metav1.ListOptions{})
-		Expect(err).NotTo(HaveOccurred(), "failed to set up watch")
+
+		listCompleted := make(chan bool, 1)
+		lw := &cache.ListWatch{
+			ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
+				options.LabelSelector = selector.String()
+				limitRanges, err := f.ClientSet.CoreV1().LimitRanges(f.Namespace.Name).List(options)
+				if err == nil {
+					select {
+					case listCompleted <- true:
+						framework.Logf("observed the limitRanges list")
+						return limitRanges, err
+					default:
+						framework.Logf("channel blocked")
+					}
+				}
+				return limitRanges, err
+			},
+			WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
+				options.LabelSelector = selector.String()
+				return f.ClientSet.CoreV1().LimitRanges(f.Namespace.Name).Watch(options)
+			},
+		}
+		_, _, w, _ := watchtools.NewIndexerInformerWatcher(lw, &v1.LimitRange{})
+		defer w.Stop()
 
 		By("Submitting a LimitRange")
 		limitRange, err = f.ClientSet.CoreV1().LimitRanges(f.Namespace.Name).Create(limitRange)
 		Expect(err).NotTo(HaveOccurred())
 
 		By("Verifying LimitRange creation was observed")
 		select {
-		case event, _ := <-w.ResultChan():
-			if event.Type != watch.Added {
-				framework.Failf("Failed to observe pod creation: %v", event)
+		case <-listCompleted:
+			select {
+			case event, _ := <-w.ResultChan():
+				if event.Type != watch.Added {
+					framework.Failf("Failed to observe limitRange creation: %v", event)
+				}
+			case <-time.After(framework.ServiceRespondingTimeout):
+				framework.Failf("Timeout while waiting for LimitRange creation")
 			}
-		case <-time.After(framework.ServiceRespondingTimeout):
-			framework.Failf("Timeout while waiting for LimitRange creation")
+		case <-time.After(10 * time.Second):
```

Comment:
why was this constant changed?  

/approve

Reply:
I think re-base auto merges. will fix it